contractcourt/htlc_incoming_contest_resolver.go

```diff
@@ -50,6 +51,19 @@ func (h *htlcIncomingContestResolver) Resolve() (ContractResolver, error) {
 		return nil, nil
 	}
 
+	// First try to parse the payload. If that fails, we can stop resolution
+	// now.
+	payload, err := h.decodePayload(h.htlc)
+	if err != nil {
+		log.Debugf("ChannelArbitrator(%v): cannot decode payload of "+
+			"htlc %v", h.ChanPoint, h.HtlcPoint())
+
+		// If we've locked in an htlc with an invalid payload on our
+		// commitment tx, we don't need to resolve it.
```

Comment:
perhaps update the comment to explain that the other party will time out the htlc if they want their funds back?

Reply:
added